[docs] reorder blog articles

return42 opened the pull request:
For details read along the [commit messages](https://github.com/searxng/searxng/pull/97/commits)

Hint: there a two patches with a fix in this PR:

- https://github.com/searxng/searxng/pull/97/commits/97d801e8a097c33a384d0b44cdbbc61a985373a6
- https://github.com/searxng/searxng/pull/97/commits/f122cb0e27dbb07cb3b5926ae7910cdc65c8fcbb

return42 committed:
[docs] remove blog article about searx-admin

- https://github.com/kvch/searx-admin last maintained 4 years ago
- searx-admin does not support 'use_default_settings' [1] (b4b81a5e)

[1] https://searxng.github.io/searxng/admin/settings.html#use-default-settings

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[docs] remove blog article 'Introducing Python 3 support'

This article is obsolete since a long time: Python 2 support has been dropped
and these days, virtualenv is managed by ``make pyenv.install``.

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[docs] move blog article "Command line engines" to admin/engines/

The article "Command line engines" should be in admin's engine
documentation (like the recoll engine).

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[docs] move blog article "SQL engines" to admin/engines/

The article "SQL engines" should be in admin's engine documentation (like the
recoll engine).

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[docs] move blog article "Offline engines" to dev/offline_engines.rst

The article "Offline engines" should be in developer's documentation next to
chapter "Engine overview".

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[docs] add engine "Demo Offline Engine"

This engine just exists for documentation purpose.

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[docs] revision of the article "Offline engines"

This patch is a a complete revision of the article "Offline engines", which also
merges the content from the searx-wiki [1] into this article.

[1] https://github.com/searx/searx/wiki/Offline-engines

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[docs] move article "XPath Engine" to to admin/engines/

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[docs] reorder chapter "Engines & Settings"

- Split chapter "Engines" and rename it into "Engines & Settings"
- Move docs/admin/engines.rst -> docs/admin/engines/engine_settings.rst

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[docs] move blog article "private engines" to admin/engines/

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[docs] add engine "Demo Online Engine"

This engine just exists for documentation purpose.

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[fix] docs/conf.py - docutils docs moved to docutils.sourceforge.io

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[fix] typo: online_dictionnary --> online_dictionary

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[docs] revision of the article "Engine Overview"

This patch revision of the article "Engine Overview":

- add links & anchors
- improve formating of the tables

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[docs] revision of the article "settings.yml"

This patch is a marginal revision of the article "settings.yml", most changes
are from normalizing the YAML syntax.

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[docs] rename 'General Engine Settings' to 'Configured Engines'

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[docs] move 'settings.yml' into admin's 'Engines & Settings'

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[docs] rename 'Recoll' to 'Recoll Engine'

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

dalf commented on docs/admin/engines/command-line-engines.rst:
```diff
@@ -59,7 +62,3 @@ This development was sponsored by `Search and Discovery Fund`_ of `NLnet Foundat
 
 .. _Search and Discovery Fund: https://nlnet.nl/discovery
 .. _NLnet Foundation: https://nlnet.nl/
-
```
"Next steps" section can be removed.

dalf commented on docs/admin/engines/private-engines.rst:
```diff
@@ -55,11 +53,5 @@ and come up with new solutions which fit your use case.
 Acknowledgement
 ===============
 
-This development was sponsored by `Search and Discovery Fund`_ of `NLnet Foundation`_ .
```
"Next steps" section can be removed.

return42 replied:
Thanks .. done .. I will push  it later .. there is a lot of other doc-stuff I am working on ATM ..

return42 committed:
[docs] revision of the blog article about local search engines

The blog article 'Query your local search engines' has been renamed 'Local
Search Engines', revised and moved into admin's chapter 'Engine & Settings'.

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[docs] move LXC blog article into developer documentation.

Move article 'Developing in Linux Containers' from blog section do developer
section.  Since there are no more articles in the blog section, remove the
section completely.

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[docs] revision of the section 'Command Line Engines'

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

dalf commented on searx/engines/demo_online.py:
```diff
@@ -0,0 +1,92 @@
+
+    """
+    global _my_online_engine  # pylint: disable=global-statement
+    _my_online_engine = engine_settings.get('name')
```
I understand it makes a demo of the `init` function, but the parameter is not used.

Out of topic for this PR, `engine_settings` values are [copied as attributes into the module](https://github.com/searxng/searxng/blob/5f76238d5cd6f9c850d5237fe639ba2900fd3e6c/searx/engines/__init__.py#L117). So `engine_settings.get('name')` could be `name` (with `global _my_online_engine, name` above).

And `engine_settings` can be:
```python
{
    k: getattr(engine, k) 
    for k in dir(engine)
    if not k.startswith('__') and not callable(getattr(engine, k))
}
```
instead of [get_engine_from_settings](https://github.com/searxng/searxng/blob/5f76238d5cd6f9c850d5237fe639ba2900fd3e6c/searx/utils.py#L488-L500)

return42 replied:
Hm, I guess I have a problem to understand why we have this argument `engine_settings` .. does it make sense to drop this argument? .. for me it was a bit confusing that we update module's name space with the values from YAML (`engine_data`):

https://github.com/searxng/searxng/blob/5f76238d5cd6f9c850d5237fe639ba2900fd3e6c/searx/engines/__init__.py#L117-L125

and later, when the processor is initialized ...

https://github.com/searxng/searxng/blob/5c5db719d2039d34aa1426bf7eb3e57055d4a54a/searx/search/processors/abstract.py#L70-L72

we call a function that gather `engine_data` from the YAML settings.

In sense of _single point of definition_ I guess it is better to say, that everything is in the namespace of engine's modul and this can be configured by (is updated from) the YAML file.

But may be I'm totally wrong, since I do not have a complete overview.

dalf commented on docs/dev/engine_overview.rst:
```diff
@@ -1,310 +1,317 @@
-The naming of overrides is arbitrary.  But the recommended overrides are the
-following:
+   Global names with a leading underline are *private to the engine* and will
+   not be overwritten.
```
Technically, names starting with `_` can be overwritten but they can be `None`.

return42 replied:
Ooops  .. I forgot to reflect your hint into the documentation.  What Do you think, is it better to fix the documentation or is it better to fix the code that overrides _private members_?

dalf replied:
I think it is better to fix the documentation.

dalf reviewed (approved):
LGTM
